File: screenshot.py
import asyncio
from playwright.async_api import async_playwright
import os

async def take_screenshots():
    # 首先检查浏览器是否已安装
    try:
        async with async_playwright() as p:
            # 尝试启动chromium
            try:
                browser = await p.chromium.launch(
                    headless=True,
                    args=['--no-sandbox', '--disable-dev-shm-usage', '--disable-gpu']
                )
            except Exception as e:
                print(f"启动浏览器失败: {e}")
                print("尝试下载 Chromium...")
                await p.chromium.download()
                browser = await p.chromium.launch(
                    headless=True,
                    args=['--no-sandbox', '--disable-dev-shm-usage', '--disable-gpu']
                )

            page = await browser.new_page(viewport={'width': 1920, 'height': 1080})

            # 登录页截图
            print('正在截取登录页...')
            await page.goto('http://localhost:3000/login', wait_until='networkidle', timeout=30000)
            await page.screenshot(path='/workspace/docs/login-page.png')
            print('登录页已保存')

            # 登录并截取首页
            print('正在登录...')
            await page.fill('input[type="text"]', '13900000001')
            await page.fill('input[type="password"]', 'user123')
            await page.click('button[type="submit"]')
            await page.wait_for_load_state('networkidle', timeout=15000)
            await asyncio.sleep(2)

            await page.screenshot(path='/workspace/docs/home-page.png')
            print('首页已保存')

            await browser.close()
            print('完成!')

    except Exception as e:
        print(f"错误: {e}")
        # 备用方案：使用curl保存HTML
        import subprocess
        subprocess.run(['curl', '-s', 'http://localhost:3000/login', '-o', '/workspace/docs/login-page.html'])
        print("已保存登录页HTML到 /workspace/docs/login-page.html")

if __name__ == '__main__':
    asyncio.run(take_screenshots())

File: screenshot.js
const puppeteer = require('puppeteer');
const path = require('path');

async function takeScreenshots() {
    const browser = await puppeteer.launch({
        headless: true,
        args: ['--no-sandbox', '--disable-setuid-sandbox']
    });

    const page = await browser.newPage();
    await page.setViewport({ width: 1920, height: 1080 });

    // 登录页截图
    console.log('正在截取登录页...');
    await page.goto('http://localhost:3000/login', { waitUntil: 'networkidle0', timeout: 30000 });
    await page.screenshot({ path: '/workspace/docs/login-page.png', fullPage: false });
    console.log('登录页已保存到 /workspace/docs/login-page.png');

    // 首页截图 - 先登录
    console.log('正在截取首页...');
    // 填写登录表单
    await page.evaluate(() => {
        const phoneInput = document.querySelector('input[type="text"]') || document.querySelector('input[placeholder*="手机"]');
        const passwordInput = document.querySelector('input[type="password"]');
        if (phoneInput) phoneInput.value = '13900000001';
        if (passwordInput) passwordInput.value = 'user123';
    });

    await page.waitForTimeout(500);

    // 提交登录
    const loginButton = await page.$('button[type="submit"]');
    if (loginButton) {
        await loginButton.click();
        await page.waitForNavigation({ waitUntil: 'networkidle0', timeout: 15000 }).catch(() => {});
    }

    // 等待页面加载
    await page.waitForTimeout(2000);

    // 截取首页
    await page.screenshot({ path: '/workspace/docs/home-page.png', fullPage: false });
    console.log('首页已保存到 /workspace/docs/home-page.png');

    await browser.close();
    console.log('截图完成!');
}

takeScreenshots().catch(console.error);
